hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/MessageMetadata.java

```diff
@@ -0,0 +1,43 @@
+package pl.allegro.tech.hermes.common.message.wrapper;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+public class MessageMetadata {
+
+    private final long timestamp;
+    private final String id;
+
+    @JsonCreator
+    public MessageMetadata(@JsonProperty("timestamp") long timestamp, @JsonProperty("id") String id) {
+        this.id = id;
+        this.timestamp = timestamp;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, timestamp);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
```

Comment:
I am aware of that. Both getClass & inheritance have some arguments but I still think that this is irrelevant here.

http://stackoverflow.com/questions/596462/any-reason-to-prefer-getclass-over-instanceof-when-generating-equals